### PR TITLE
Support editable breadcrumb

### PR DESCRIPTION
**Pull request by zhx828**
I propose editable breadcrumb for Reference Genome, Genomic Change and HGVSg. I'm also aware of google firebase breadcrumb design and think it makes more sense if we also want to support changes to protein change. We can have a separate pull request to support that. This fixes https://github.com/oncokb/oncokb/issues/3487

In this PR, I created four types of breadcrumbs:
- TextBreadcrumb
Screen Shot 2023-07-12 at 5 09 13 PM

- LinkBreadcrumb
- InputBreadcrumb
Screen Shot 2023-07-12 at 4 40 19 PM

- DropdownBreadcrumb
Screen Shot 2023-07-12 at 4 22 17 PM

This is how it looks like when a HGVSg or Genomic Change is specified
Screen Shot 2023-07-12 at 4 22 12 PM

### CODE REVIEW

**Comment by calvinlu3** [comment]
The different breadcrumbs look pretty good to me. Just a couple of questions:
- Should we also support `LinkBreadcrumb` that can be edited. For instance, might want to change the gene or go back to the gene page.
- Will we also have dropdown breadcrumbs that can be searchable?

File: src/main/webapp/app/store/AnnotationStore.ts [hunk 1]
```diff
@@ -97,55 +101,7 @@ export class AnnotationStore {
...
-    return this.mutationMapperStore.result
-      ? findPositionFilter(
-          this.mutationMapperStore.result.dataStore.selectionFilters
-        )
-      : undefined;
-  }
-
-  @computed get selectedPositions() {
-    return this.positionFilter ? this.positionFilter.values : [];
-  }
```
Comment (zhx828) [inline src/main/webapp/app/store/AnnotationStore.ts:148]: I restructured the content so all computed values are there. I will mention the portion that you need to review.

File: src/main/webapp/app/store/AnnotationStore.ts [hunk 2]
```diff
@@ -524,6 +451,158 @@ export class AnnotationStore {
...
+          counts: datum.length,
+        });
+        return acc;
+      },
+      [] as Oncogenicity[]
+    );
+  }
+
+  @computed
+  get annotationData() {
```
Comment (zhx828) [inline src/main/webapp/app/store/AnnotationStore.ts:485]: This is new

File: src/main/webapp/app/store/AnnotationStore.ts [hunk 3]
```diff
@@ -524,6 +451,158 @@ export class AnnotationStore {
...
+          : {
+              alteration: this.alteration.result.alteration,
+              name: this.alteration.result.name,
+            },
+        true
+      );
+    } else {
+      return this.annotationData.result.query.alteration || '';
+    }
+  }
```
Comment (zhx828) [inline src/main/webapp/app/store/AnnotationStore.ts:580]: This section is updated.

File: src/main/webapp/app/pages/annotationPage/AnnotationPage.tsx [hunk 4]
```diff
@@ -321,112 +337,189 @@ export default class AnnotationPage extends React.Component<
...
+                  refGenome: selectedOption.value as REFERENCE_GENOME,
+                  cancerType: this.props.store.cancerTypeName,
+                })
+              );
+            },
+            options: [REFERENCE_GENOME.GRCh37, REFERENCE_GENOME.GRCh38],
+          } as IDropdownBreadcrumb,
+          {
+            type: 'input',
+            key: 'hgvgs',
```
Comment (calvinlu3) [inline src/main/webapp/app/pages/annotationPage/AnnotationPage.tsx:426]:
```suggestion
            key: 'hgvsg',
```

**Review by calvinlu3 — APPROVED** [review]
LGTM

**Pull request merged**